File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroTT(vector<vector<int>>& matrix, int ii, int jj){
        int n= matrix.size();
        int m= matrix[0].size();
        for(int i=0; i<n; i++){
            // matrix[ii][i]=0;
            matrix[i][jj]=0;
        }
        for(int i=0; i<m; i++){
            matrix[ii][i]=0;
        }
    }
    void setZeroes(vector<vector<int>>& matrix) {
        int n= matrix.size();
        int m= matrix[0].size();
        vector<pair<int, int>>pairs;
        for(int i=0; i<n; i++){
            for(int j=0; j<m; j++){
                if(matrix[i][j]==0){
                    pairs.push_back(make_pair(i,j));
                }
            }
        }
        for(auto x: pairs){
            int i= x.first;
            int j= x.second;
            setZeroTT( matrix, i, j);
        }
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int n= matrix.length;
        int m= matrix[0].length;
        ArrayList<Integer> r= new ArrayList<>();
        ArrayList<Integer> c= new ArrayList<>();
        for(int i=0; i< n; i++){
            for(int j=0; j< m; j++){
                if(matrix[i][j]==0){
                    r.add(i);
                    c.add(j);
                }
            }
        }
        // for(int ii=0; ii< r.size(); ii++){
        //     int rx= r.get(ii);
        //     for(int i=0; i< matrix[rx].length; i++){
        //         matrix[rx][i]=0;
        //     }
        // }
        for(int i=0; i< r.size(); i++){
            int rx= r.get(i);
            for(int j=0; j< m; j++){
                matrix[rx][j]=0;
            }
        }
        for(int i=0; i< c.size(); i++){
            int cx= c.get(i);
            for(int j=0; j< n; j++){
                matrix[j][cx]=0;
            }
        }
    }
}